File: HackerEarth/Data Structure/Stacks/ArrayFormation.cpp
#include <bits/stdc++.h>
using namespace std;
 
#define fastio ios_base::sync_with_stdio(false); cin.tie(NULL); cout.tie(NULL);
 
bool isPrime(long int val)
{
   if(val <= 1)return false;
 
   for(long int i=2; i<=sqrt(val); i++){
      if(val%i == 0)return false;
   }
 
   return true;
}
 
 
int main()
{
    fastio;
    long int n;
    queue<long int>arr;
    queue<long int>prime;
    stack<long int>composite;
    cin >> n;
 
    for(int i=0; i<n; i++){
        long int temp;
        cin >> temp;
        arr.push(temp);
    }
 
    while(!arr.empty()){
        long int val = arr.front();
        if(isPrime(val))prime.push(val);
        else composite.push(val);
        arr.pop();
    }
 
    while(!prime.empty()){
        cout << prime.front() << " ";
        prime.pop();
    }
    cout << "\n";
    while(!composite.empty()){
        cout << composite.top() << " ";
        composite.pop();
    }
 
 
    return 0;
}

File: HackerEarth/Data Structure/Stacks/ArrayFormation.py
# 1 test case failed
def isPrime(n) :
    if (n <= 1) :
        return False
    if (n <= 3) :
        return True
 
    if (n % 2 == 0 or n % 3 == 0) :
        return False
 
    i = 5
    while(i * i <= n) :
        if (n % i == 0 or n % (i + 2) == 0) :
            return False
        i = i + 6
 
    return True
 
n = int(input())
arr = list(map(int, input().split()))
 
stack = []
queue = []
for i in arr:
    if isPrime(i):
        queue.append(i)
    else:
        stack.append(i)
 
for i in queue:
    print(i, end = " ")
print()
stack = reversed(stack)
for i in stack:
    print(i, end = " ")
